src/platform/STM32/bus_i2c_stm32f4xx.c

```diff
@@ -116,6 +116,14 @@ const i2cHardware_t i2cHardware[I2CDEV_COUNT] = {
 
 i2cDevice_t i2cDevice[I2CDEV_COUNT];
 
+// State used by event handler ISR
+typedef struct {
+    uint8_t subaddress_sent;    // flag to indicate if subaddess sent
+    uint8_t final_stop;         // flag to indicate final bus condition
+    int8_t index;               // index is signed -1 == send the subaddress
+} i2cEvState_t;
+static i2cEvState_t i2c_ev_state[I2CDEV_COUNT];
```

Comment:
Probably better to make the compiler aware to not optimize because it's part of the ISR
```suggestion
static volatile i2cEvState_t i2c_ev_state[I2CDEV_COUNT];
```

Reply:
This doesn’t need to be volatile. And we need the ISR to be as optimal as possible!

Reply:
@KarateBrot: https://www.kernel.org/doc/Documentation/process/volatile-considered-harmful.rst

Reply:
I know that an ISR should optimally take no runtime at all. I was contemplating for a long time if I should post this change request because I already had a weird feeling about it. But I can now clearly see why `volatile` is not needed here and I feel like an idiot, sorry.